Stabilize the tag filter component and adjust for screen reader compa…

tristandasilva opened the pull request:
Related to [WET-468](https://jtickets.atlassian.net/browse/WET-468)

tristandasilva committed:
Stabilize the tag filter component and adjust for screen reader compatibility

Garneauma reviewed (changes requested):
@tristandasilva This is great work!

Can you try to fix the status message for the Filter plugin? It seems to not say the number of items correctly, which impacts the accessibility of the tagfilter plugin when they are implemented together.

Also, can you add some descriptions of the tests you performed when doing your accessibility assessment? Only for the most relevant success criteria like Info and relationship, Meaningful sequence, Keyboard, Headings and labels, Status messages.

Garneauma commented on src/plugins/filter/filter.js:
```diff
@@ -226,8 +238,8 @@ var componentName = "wb-filter",
-				if ( hndParentSelector ) {
-					$item = $item.parentsUntil( hndParentSelector );
+				if ( hdnParentSelector ) {
+					$item.parentsUntil( hdnParentSelector ).addClass( filterClass );
```
Was something broken here? It seems that, before, the class was only added to the parent element, now the class is added to the item and the parent?

Garneauma commented on src/plugins/tagfilter/tagfilter.js:
```diff
@@ -135,45 +148,35 @@ const componentName = "wb-tagfilter",
+						if ( selectedItem && selectedItem.value !== "" ) {
+							selectedFilters.push( selectedItem.value );
+						} else {
+							console.warn( componentName + ": Radio button groups must have a default selected value. If you want to display all items, add an option called \"All\" with an empty value." );
```
This doesn't seem to work. I get the warning for most of the examples now. The "if" statement seems wrong. The "All" item must have a value of "", so this wouldn't work.

Garneauma commented on src/plugins/tagfilter/tagfilter.js:
```diff
@@ -212,50 +214,67 @@ const componentName = "wb-tagfilter",
-					} ).length;
-
-					if ( itemIncludesFilter ) {
+			let matchCount = 0;
```
For code consistency, we try to keep all variable declarations within a block together. Please move back to where it was.

tristandasilva commented:
@Garneauma I reverted the timeout duration for the "keyup" event to 250, and added a timeout when updating the status message (line 278 in filter.js) as you suggested and it works well!

Garneauma reviewed (changes requested):
Accessibility messages seem to be working very well! Only a few small changes.

Yves-Denis reviewed (approved):
Security Assessment Completed - Passed, No concerns identified
